domain/ide/workspace/folder.go

```diff
@@ -417,41 +360,113 @@ func sendAnalytics(data *snyk.ScanData) {
-	scanEvent.Data.Attributes.Status = "Success"
-	scanEvent.Data.Attributes.ScanType = string(data.Product)
-	scanEvent.Data.Attributes.UniqueIssueCount.Critical = data.SeverityCount[data.Product].Critical
-	scanEvent.Data.Attributes.UniqueIssueCount.High = data.SeverityCount[data.Product].High
-	scanEvent.Data.Attributes.UniqueIssueCount.Medium = data.SeverityCount[data.Product].Medium
-	scanEvent.Data.Attributes.UniqueIssueCount.Low = data.SeverityCount[data.Product].Low
-	scanEvent.Data.Attributes.DurationMs = fmt.Sprintf("%d", data.DurationMs)
-	scanEvent.Data.Attributes.TimestampFinished = data.TimestampFinished
-
-	bytes, err := json.Marshal(scanEvent)
+	ic := gafanalytics.NewInstrumentationCollector()
+
+	ua := util.GetUserAgent(gafConfig, config.Version)
+	ic.SetUserAgent(ua)
+
+	iid := instrumentation.AssembleUrnFromUUID(uuid.NewString())
+	ic.SetInteractionId(iid)
+
+	ic.SetTimestamp(data.TimestampFinished)
+	ic.SetDuration(data.DurationMs)
+
+	ic.SetStage("dev")
+	ic.SetType("analytics")
+	ic.SetInteractionType("Scan done")
+
+	categories := setupCategories(data, c)
+	ic.SetCategory(categories)
+
+	ic.SetStatus(gafanalytics.Success)
+
+	summary := createTestSummary(data, c)
+	ic.SetTestSummary(summary)
+
+	targetId, err := instrumentation.GetTargetId(data.Path, instrumentation.AutoDetectedTargetId)
+	if err != nil {
+		logger.Err(err).Msg("Error creating the Target Id")
+	}
+	ic.SetTargetId(targetId)
+
+	ic.AddExtension("device_id", c.DeviceID())
+
+	analyticsData, err := gafanalytics.GetV2InstrumentationObject(ic)
 	if err != nil {
-		logger.Err(err).Msg("Error marshaling scan event")
+		logger.Err(err).Msg("Error creating the instrumentation collection object")
 		return
 	}
 
-	err = analytics.SendAnalyticsToAPI(c, bytes)
+	v2InstrumentationData, err := json.Marshal(analyticsData)
 	if err != nil {
-		logger.Err(err).Msg("Error sending analytics to API")
+		logger.Error().Err(err).Msg("Failed to marshal analytics")
+	}
+
+	logger.Debug().Any("v2InstrumentationData", string(v2InstrumentationData)).Msg("Analytics data")
+
+	err = analytics.SendAnalyticsToAPI(c, v2InstrumentationData)
+	if err != nil {
+		logger.Err(err).Msg("Error sending analytics to API: " + string(v2InstrumentationData))
 		return
 	}
 }
 
+func setupCategories(data *snyk.ScanData, c *config.Config) []string {
+	args := []string{product.ToProductCodename(data.Product), "test"}
+	args = append(args, c.CliSettings().AdditionalOssParameters...)
+	categories := instrumentation.DetermineCategory(args, c.Engine())
+	return categories
+}
+
+func createTestSummary(data *snyk.ScanData, c *config.Config) json_schemas.TestSummary {
+	logger := c.Logger().With().Str("method", "folder.createTestSummary").Logger()
+	sic := data.GetSeverityIssueCounts()
+	testSummary := json_schemas.TestSummary{Type: string(data.Product)}
+
+	if len(sic) == 0 {
+		logger.Info().Msgf("skipping create test summary. no issues found for product %v", string(data.Product))
```

Comment:
Actually, we are not skipping it, we are just returning it with 0s :). Which is what is required & expected :)

Reply:
Also, I'd put that to debug mode.

Reply:
Yup agree 
I will change it to "no scan issues found for product x"